app/models/lightweight_activity.rb

```diff
@@ -154,7 +154,28 @@ def export
     self.plugins.each do |p|
       activity_json[:plugins] << p.export
     end
-    # TODO: add special code to push glossary plugin onto plugin list if glossary is set
+
+    # if the activity has a glossary model assiged to it, add the fake glossary plugin to the list of plugins
+    # replacing any existing glossary plugins and using the existing glossary plugin's approved script
+    if self.glossary_id && approved_glossary_script = ApprovedScript.find_by_label("glossary")
+      # remove any existing glossary script
+      activity_json[:plugins].delete_if { |plugin| plugin[:component_label] == "glossary" }
+
+      fake_glossary_plugin = {
+        id: 0,
```

Comment:
A sequence could have multiple activities each with their own glossary.  
If this is loaded in the AP I suspect if they all have the same id, it might cause problems.

Also I'm not sure, but perhaps this plugin id is used when the glossary plugin stores learner data. If the id has to be a number, I suggest using `glossary_id*1_000_000_000+activity_id`. That would work until we have more than a 1 billion activities and 9 billion glossaries. I'm assuming we can use int64 for these numbers. It is hacky, string ids would be better, but that might require changes to the AP.  

I don't remember what we do with glossaries shared across sequences. If we support that (or need to) that would affect this. I think they currently don't share the student data. There might be a story about this scheduled for later.

Reply:
We fixed this in a commit in a newer PR, as it also changes the spec to test the ids whereas the ids are not tested in this commit.  You can see the changes here:

https://github.com/concord-consortium/lara/pull/922/commits/dea6094845a1de48165972c38a33b008fd64360a

Reply:
sounds good